### ISSUE TITLE
Set-HPOVApplianceNetworkConfig does not run in async mode

### ISSUE THREAD

**Issue by VincentBerger** [HewlettPackard/POSH-HPOneView#295]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Set-HPOVApplianceNetworkConfig should honor the -Async parameter. It would be useful when setting up multiple appliances (and this cmdlet supports only 1 ApplianceConnection at a time)

### Actual Behavior
Set-HPOVApplianceNetworkConfig always waits for the task completion before returning

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`): 3.10.1416.3042
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  
Output from `$PSVersionTable` on your Windows Host:

```PowerShell
Name                           Value
----                           -----
PSVersion                      5.1.14393.1480
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.1480
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by ChrisLynchHPE** [comment]
The Cmdlet doesn't have an `-Async` parameter.  From a UX perspective, its quite hard to support that model.  It has to be blocking while the configure network task runs and doesn't fail to scenarios like duplicate IP address detected. 

The way the Cmdlet works is that you provide the network configuration.  The address you used to connect to the appliance with `Connect-HPOVMgmt` is used to send the request.  The Cmdlet waits 90 seconds, and then if the configuration was to set a Static IPv4 address and it didn't match what was in the `Connect-HPOVMgmt` call, it attempts to use the new address to see if the target IP address is a OneView appliance.  If that connection fails, it then attempts to connect to the original address to locate the task results.  If it is a OneView appliance, it updates the `HPOneView.Library.ApplianceConnection` object in the `$ConnectedSessions` global variable, then attempts to get the task results using the new address.

**Comment by VincentBerger** [comment]
I'll readily admit my use case is special (prepare many DCS appliances for Discover living labs), but it would be nice to have an -Async switch that skips the waiting logic and just returns the task, so that I can start 10 in parallel and wait for them all

**Comment by ChrisLynchHPE** [comment]
I would suggest you use Runspaces.  Take a look at [this article](https://blogs.technet.microsoft.com/heyscriptingguy/2015/11/28/beginning-use-of-powershell-runspaces-part-3/) on how to use them, so you can multi-thread your script.  If I get time, I can help write up an example on how to accomplish this task.

**Comment by VincentBerger** [comment]
Good suggestion, thanks. I managed to make it work, 10 appliances configured in parallel. I found it a bit clunky to debug or get output from the threads, but it's working...

**Comment by ChrisLynchHPE** [comment]
Closing due to no further activity.
